src/dune_rules/module_compilation.ml

```diff
@@ -230,6 +243,75 @@ let build_cm cctx ~precompiled_cmi ~cm_kind (m : Module.t)
     >>| Action.Full.add_sandbox sandbox))
   |> Memo.Option.iter ~f:Fun.id
 
+let build_melange_js ~pkg_name ~module_system ~dst_dir ~cctx m =
```

Comment:
This function has little to do with regular compilation, so you can move it to `Melange_rules`.

Reply:
https://github.com/ocaml/dune/pull/6349/commits/677ae55afcaad5e60d274a620868cdd6e98bc696